opm/parser/eclipse/EclipseState/SimulationConfig/HysteresisOptions.cpp

```diff
@@ -0,0 +1,230 @@
+#include <opm/parser/eclipse/EclipseState/SimulationConfig/HysteresisOptions.hpp>
+#include <opm/parser/eclipse/Utility/String.hpp>
+
+namespace Opm {
+
+namespace {
+
+HysteresisOptions::Model readHysteresisModel( int x ) {
+    using Model = HysteresisOptions::Model;
```

Comment:
Is the `Model::` part mandatory (i.e., a enum class)? IMO this will look weird in downstream code where a model usually is the code which discretizes the PDEs...